File: suites/osprey-compute-kernels/reference/cases/factorial/factorial.hs
-- Factorial-style product 1*2*...*N taken mod 1000000007 (matches factorial.osp).
import Data.List (foldl')

modP :: Int
modP = 1000000007

step :: Int -> Int -> Int
step acc i = (acc * i) `mod` modP

main :: IO ()
main = print (foldl' step 1 [1 .. 10000000 :: Int])

File: suites/osprey-compute-kernels/reference/cases/factorial/factorial.rs
// Factorial-style product 1*2*...*N taken mod 1000000007 (matches factorial.osp).
const MOD: i64 = 1000000007;

fn main() {
    let mut acc: i64 = 1;
    let mut i: i64 = 1;
    while i <= 10000000 {
        acc = (acc * i) % MOD;
        i += 1;
    }
    println!("{}", acc);
}
